File: Models/Product.cs
using dotnet_backend_freshmart.Models.Enums;

namespace dotnet_backend_freshmart.Models
{
    public class Product
    {
        public Guid Id { get; set; }

        public string Sku { get; set; } = string.Empty;

        public string? Barcode { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public Guid CategoryId { get; set; }

        public Guid? SupplierId { get; set; }

        public string Unit { get; set; } = "Cai";

        public decimal CostPrice { get; set; } = 0;

        public decimal SellPrice { get; set; } = 0;

        public decimal VatRate { get; set; } = 8.00m;

        public int Stock { get; set; } = 0;

        public int MinStock { get; set; } = 0;

        public string? ImageUrl { get; set; }

        public StockStatus Status { get; set; } = StockStatus.InStock;

        public bool IsActive { get; set; } = true;

        public DateTime? ExpiryDate { get; set; }

        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Khóa ngoại liên kết tới Danh mục (Category)
        public virtual Category? Category { get; set; }

        // Khóa ngoại liên kết tới Nhà cung cấp
        public virtual Supplier? Supplier { get; set; }
    }
}

File: freshmart/src/types/product.ts
// ============================================================================
// Types cho module Sản phẩm (Products)
// Khớp 100% với DTOs của Backend ASP.NET Core
// ============================================================================

export type StockStatus = 'InStock' | 'LowStock' | 'OutOfStock';

export interface Product {
  id: string;
  sku: string;
  barcode?: string | null;
  name: string;
  description?: string;
  categoryId: string;
  categoryName: string;
  categorySlug: string;
  supplierId?: string | null;
  supplierName?: string | null;
  unit: string;
  costPrice: number;
  sellPrice: number;
  vatRate: number;
  stock: number;
  minStock: number;
  imageUrl?: string | null;
  status: StockStatus;
  isActive: boolean;
  expiryDate?: string | null;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProductRequest {
  sku: string;
  barcode?: string | null;
  name: string;
  description?: string;
  categoryId: string;
  supplierId?: string | null;
  unit?: string;
  costPrice: number;
  sellPrice: number;
  vatRate?: number;
  stock: number;
  minStock: number;
  imageUrl?: string | null;
  status?: StockStatus;
  isActive?: boolean;
  expiryDate?: string | null;
  notes?: string | null;
}

export interface UpdateProductRequest {
  sku: string;
  barcode?: string | null;
  name: string;
  description?: string;
  categoryId: string;
  supplierId?: string | null;
  unit?: string;
  costPrice: number;
  sellPrice: number;
  vatRate?: number;
  stock: number;
  minStock: number;
  imageUrl?: string | null;
  status?: StockStatus;
  isActive?: boolean;
  expiryDate?: string | null;
  notes?: string | null;
}

export interface ProductFilterParams {
  search?: string;
  categoryId?: string;
  categorySlug?: string;
  supplierId?: string;
  status?: StockStatus;
  isActive?: boolean;
  sortBy?: 'name_asc' | 'name_desc' | 'price_asc' | 'price_desc' | 'stock_asc' | 'stock_desc' | 'newest';
  page?: number;
  limit?: number;
}

export interface QuickStockRequest {
  stock: number;
  note?: string;
}
